### ISSUE TITLE
Salt isn't random

### ISSUE THREAD

**Issue by joeystevens00** [smallwat3r/shhh#29]
The salt isn't randomly generated but rather it's the first 16 chars of the secret provided by the user.  If the secret is less than 17 characters then the salt is just the secret. I'm no cryptographer but generally salts are randomly generated instead of being derived from the secret itself.
https://github.com/smallwat3r/shhh/blob/master/shhh/encryption.py#L53

**Comment by smallwat3r** [comment]
Thanks @joeystevens00

The random salt is generated [here](https://github.com/smallwat3r/shhh/blob/d38a10f1e57ba241421e4824c9d5551f25d7a301/shhh/encryption.py#L38) using `secrets.token_bytes(16)` it randomly creates a salt of 16 chars.

Shhh is using the passphrase to encrypt the value, instead of a secret key [as seen here](https://github.com/smallwat3r/shhh/blob/d38a10f1e57ba241421e4824c9d5551f25d7a301/shhh/encryption.py#L34).

I also have to include the salt and the HMAC iteration in the message when encrypting. To decrypt the message I then first have to separate those values to get the encrypted value to be Fernet-compatible.

I saw this from Martin Peters on StackOverflow "Fernet with password" which makes me use this technic [here](https://stackoverflow.com/a/55147077/8705066) when I switched the encrypt method to Fernet.

Hope this answers your question?
